src/js/controller/instream-adapter.js

```diff
@@ -207,8 +207,14 @@ const InstreamAdapter = function(_controller, _model, _view, _mediaPool) {
         adModel.set('state', newstate);
 
         if (newstate === STATE_PLAYING) {
+            if (event.playReason) {
+                _model.set('playReason', event.playReason);
+            }
             _controller.trigger(AD_PLAY, event);
         } else if (newstate === STATE_PAUSED) {
+            if (event.pauseReason) {
+                _model.set('pauseReason', event.pauseReason);
+            }
```

Comment:
instream should not be set player model state. These should be adModel play and pause reasons if we wanted this state in the adModel. We do not want to attribute the reason for content playback to the reason for ad playback after an ad.

These two models are used to track state. Not simply as event dispatchers. If you want to call `changeReason` in the controller when something happens here in instream, then you should trigger events from instream with handlers in the controller. The good thing is we already trigger events on the controller:

With that in mind, we can listen for `AD_PLAY` and `AD_PAUSE` in the controller instead of adding changes to the player model. In the controller this would look like:

```js
// this can be one line
_model.on('change:pauseReason change:playReason', changeReason);
// listen for play and pause reasons from instream
_this.on(AD_PLAY, (event) => { changeReason(null, event.playReason); });
_this.on(AD_PAUSE, (event) => { changeReason(null, event.pauseReason); });
```